docs/contributing.md

```diff
@@ -0,0 +1,26 @@
+# Contributing
+
+OpenSAFELY is an open source project; all our code is available for inspection and reuse, and we can accept contributions in the following ways:
+
+## Documentation
+
+Typos and other small improvements are always very welcome. Raise an issue, or even better a Pull Request, in the [documentation repo](https://github.com/opensafely/documentation/). We will incorporate these contributions after we've checked them for consistency with our house style.
+
+We welcome proposals for longer topic-based guides. Suggest them as an [issue in the documentation repo](https://github.com/opensafely/documentation/issues), and we'll work with you to define an overall structure and format, before you write and submit the draft.
+
+## Reusable actions
+These are units of software that solve a problem for several studies without the need to copy-and-paste between them, [described in more detail here](./actions-reusable.md).  They can be shared between researchers, even between groups that use different programming languages, and are one of the best ways you can make contributions that benefit the community. If you've written a reusable action you'd like to contribute to the actions library, please get in touch at [team@opensafely.org](mailto:team@opensafely.org).
+
+## Study definition patterns
+[Study definitions](./study-def.md) are intended to be shared under open source licences, and they can be found in the [OpenSAFELY Github organisation](https://github.com/opensafely).  We welcome all ideas that may be able to help other members of the community, which we can incorporate in our documentation ([for example on this page about study definition tips](https://docs.opensafely.org/study-def-tricks/)); we're also happy to help publicise blog posts, or host them as guest blog posts on our website. Just submit a ticket to the [documentation repository](https://github.com/opensafely/documentation/issues) with your suggestion.
+
+## Peer support
+
+We encourage researchers to post questions in the [Q&A Forum](https://github.com/opensafely/documentation/discussions). We would love more people to chip in and attempt to answer questions!
+
+## Core code
+We would like to accept _ad hoc_ contributions to the [core code](https://github.com/opensafely-core/), but this is relatively hard for us to do. New features and bugfixes need a lot work from our side to integrate and maintain, because a lot of the design and prioritisation decisions are made with lots of different parts of the framework in mind. All but the most trivial changes also come with intellectual property considerations, so we also need to secure Contributor Agreements from individuals or their employers.  We plan to seek funding to help resource this kind of work, as we believe it can really strengthen and build our community, but for the time being, we suggest [filing bug reports or feature requests](https://github.com/opensafely-core) rather than contributing code.
+
```

Comment:
> All but the most trivial changes also come with intellectual property considerations, so we also need to secure Contributor Agreements from individuals or their employers.

I wonder about this line.  It doesn't sound very open source to be talking about intellectual property.  Does it need to be mentioned at all?

Reply:
I don't think it hurts to mention the topic - as the paragraph at the moment is just giving a few template justifications why we might need to ignore an "easy to merge" external patch. I think you have a point about avoiding the buzzphrase "intellectual property" though, as some people might bridle at it - maybe something more vague like "...also come with legal risks...".

Reply:
You could also switch out:

> All but the most trivial changes also come with intellectual property considerations…

for

> Most code changes have an associated administrative and legal burden: we need to secure contributor agreements from individual or their employers.